frontend/src/components/Arrivals.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './Arrivals.css';
+
+const Arrivals = ({ tubeData, durationPassed }) => {
+  const [newArrival, setNewArrival] = useState('');
+  const durationPassedRef = useRef(durationPassed);
+  
+  useEffect(() => {
+    
+    const scheduleTrains = () => {
+      const trainData = tubeData;
+      const arrivingNext = [];
+      if (trainData.length > 0) { 
+        if (durationPassedRef.current > trainData[0].timeToStation) {
```

Comment:
It would be great to talk this through with you once more maybe in the group as I think i've got most of it now but trying to understand this line- if duration passed is greater than timeToStation, so its already been 10 seconds and the timeToStation was 5 seconds, then we are dealing with an arrival that should have already happened? Is that why we are therefore scheduling it to arrive, as its already late as it were?